src/Moryx.Products.UI.Interaction/Recipes/Details/RecipeDetailsViewModelBase.cs

```diff
@@ -81,16 +81,19 @@ public async Task Load(RecipeViewModel recipeVm, IReadOnlyCollection<WorkplanVie
             await LoadTypeAndWorkplans(recipeVm.Model, workplans);
 
             WorkplanViewModel wp = null;
-            if (recipeVm.Model.WorkplanId != 0)
-                wp = workplans.FirstOrDefault(w => w.Id == recipeVm.Model.WorkplanId);
+            if (recipeVm.Model.WorkplanModel.Id != 0)
+                wp = Workplans.FirstOrDefault(w => w.Id == recipeVm.Model.WorkplanModel.Id && w.Version == recipeVm.Model.WorkplanModel.Version);
             recipeVm.Workplan = wp;
 
             EditableObject = recipeVm;
         }
 
         private async Task LoadTypeAndWorkplans(RecipeModel model, IReadOnlyCollection<WorkplanViewModel> workplans)
         {
-            Workplans = workplans;
+            if (workplans.Any(w => w.Id == model.WorkplanModel.Id && w.Version == model.WorkplanModel.Version))
```

Comment:
Some as above. Consider null value, compare by id alone and concat model when necessary and available.

The idea is: An updated server sends updated data, that the UI can use to fix the behavior. But in that case the client does not need a new version.